docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,197 @@
+
+### Operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By successfully initiating the file, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must be included in each operation like upload the blocks, multipart upload completion request of a file.
+If the file has single block then the operator emits an empty string, this is an indication of not to use the multi-part feature for the downstream operators.
+    -   This operator emits the **(filemetadata, uploadId)** to **S3FileMerger** and **(filePath, metadata, uploadId)** to **S3BlockUpload**.
+
+-   ***S3BlockUploadOperator***
+    -   **S3BlockUpload** operator upload the blocks into S3 using different API's which depends on number of blocks of a file.
+If the file has single block then upload the block into S3 using **AmazonS3Client.putObject(...)** API. S3 returns a response of type **PutObjectResult** includes the **ETag**.
+If the file has more blocks then upload the block using **AmazonS3Client.uploadPart(...)** method. S3 returns a response of type UploadPartResult includes the **ETag**. This **ETag** value must be include in request to complete multipart upload.
+    -   **S3BlockUpload** emits **(path, ETag)** to **s3FileMerger**.
+
+-   ***S3FileMerger***
+    -   Complete multipart upload request using **AmazonS3Client.completeMultipartUpload(...)** and must include the **upload ID**, a list of both part numbers and corresponding **ETag** values. **S3FileMergerOperator** send the complete multi-part upload request to S3 once it have all the **part ETag's** of a file. 
+**Amazon S3** creates an object by concatenating the parts in ascending order based on part number. After a successful upload request, the parts no longer exist and S3 response includes an **ETag** which uniquely identifies the combined object data. 
+
+### Configuration Parameters
+-----------------------------
+
+-   ***accessKey*** -   String
+    -   Mandatory Parameter    
+    -   Specifies the AWS access key to access Amazon S3 and has permissions to access the specified bucket.
+    -   Example value = AKIAJVAGFANC2LSZCJ4Q
+
+-   ***secretAccessKey***   -   String
+    -   Mandatory Parameter
+    -   Specifies the AWS secret access key to access Amazon S3 and has permissions to access the specified bucket.
+    -   Example value = wpVr3U82RmCKJoY007YfkaawT7CenhTcK1B8clue
+    
+-   ***endPoint***  -   String
+    -   Endpoint is the url that is entry point for a web service. Specify the valid endpoint to access S3 bucket.
+    -   Example value = s3.amazonaws.com
```

Comment:
I don't see this property used in the example; is this the default value ? If so, add a note here to that effect. If not, explain how this value is set in the example.


Also, verify if the scheme prefix like "https://" is necessary or optional or not valid.